diff -ru a/qutebrowser/mainwindow/tabbedbrowser.py b/qutebrowser/mainwindow/tabbedbrowser.py
--- a/qutebrowser/mainwindow/tabbedbrowser.py	2019-12-05 10:32:05.000000000 -0500
+++ b/qutebrowser/mainwindow/tabbedbrowser.py	2020-01-08 20:45:25.565768631 -0500
@@ -491,6 +491,11 @@
         if (config.val.tabs.tabs_are_windows and self.widget.count() > 0 and
                 not ignore_tabs_are_windows):
             window = mainwindow.MainWindow(private=self.is_private)
+            role_background = background if background is not None else config.val.tabs.background
+            if role_background:
+                window.setWindowRole('qutebrowser_background')
+            else:
+                window.setWindowRole('qutebrowser')
             window.show()
             tabbed_browser = objreg.get('tabbed-browser', scope='window',
                                         window=window.win_id)
