diff --git a/nix-top b/nix-top
index 9577764..3db5572 100755
--- a/nix-top
+++ b/nix-top
@@ -3,6 +3,7 @@
 require "shellwords"
 require "optparse"
 require "io/wait"
+require "etc"
 
 DEFAULTS = {
   delay: 0.5,
@@ -35,14 +36,26 @@ def size
   `echo 'cols\nlines' | tput -S`.strip.split("\n").map(&:to_i)
 end
 
-def build_users
-  `getent group nixbld`.strip.split(":").last.split(",")
+$builds_dir = "/nix/var/nix/builds"
+
+def build_dirs()
+  builds_dir = "/nix/var/nix/builds"
+  Dir.children(builds_dir).map do |name|
+    dir = "#{builds_dir}/#{name}"
+    if File.exist?("#{dir}/env-vars")
+      dir
+    elsif File.exist?("#{dir}/build/env-vars")
+      "#{dir}/build"
+    else
+      nil
+    end
+  end.reject(&:nil?)
 end
 
 def active_build_users()
-  build_users.select do |name|
-    `pgrep -fu #{name.shellescape}`
-    $? == 0
+  build_dirs.map do |dir|
+    file_uid = File.stat(dir).uid
+    Etc.getpwuid(file_uid).name
   end
 end
 
@@ -66,16 +79,14 @@ end
 # Gets either the out path, or an approximation, depending whether
 # or not it can peek into processes.
 def get_out_path(user, pid)
-  begin
-  # whew!
-  build_dir = `find -L /nix/var/nix/builds -maxdepth 2 -user #{user.shellescape} -exec stat --printf '%Z:%n\\n' '{}' ';' | sort -n`
-    .strip
-    .split("\n")
-    .last
-    .split(":")
-    .last
-  rescue
-    "(unknown)"
+  build_dir = "(unknown)"
+  for dir in build_dirs do
+    file_uid = File.stat(dir).uid
+    file_user = Etc.getpwuid(file_uid).name
+    if file_user == user
+      build_dir = dir
+      break
+    end
   end
   begin
     file = "#{build_dir}/env-vars"
