diff --git a/qutebrowser/mainwindow/tabbedbrowser.py b/qutebrowser/mainwindow/tabbedbrowser.py
index 5110a2d69..9b3dbaa84 100644
--- a/qutebrowser/mainwindow/tabbedbrowser.py
+++ b/qutebrowser/mainwindow/tabbedbrowser.py
@@ -638,6 +638,10 @@ class TabbedBrowser(QWidget):
 
         prev_focus = QApplication.focusWidget()
 
+        if background:
+            objects.qapp.setDesktopFileName('qutebrowser_background')
+        else:
+            objects.qapp.setDesktopFileName('qutebrowser')
         if config.val.tabs.tabs_are_windows and self.widget.count() > 0:
             window = mainwindow.MainWindow(private=self.is_private)
             tab = window.tabbed_browser.tabopen(
