diff --git a/css/filmstrip/_tile_view.scss b/css/filmstrip/_tile_view.scss
index ecc651d26..e80ec137b 100644
--- a/css/filmstrip/_tile_view.scss
+++ b/css/filmstrip/_tile_view.scss
@@ -108,7 +108,7 @@
      * Max-width corresponding to the ASPECT_RATIO_BREAKPOINT from features/filmstrip/constants,
      * from which we subtract the chat size.
      */
-    @media only screen and (max-width: calc(500px + #{$sidebarWidth})) {
+    @media only screen and (max-width: calc(100px + #{$sidebarWidth})) {
         video {
             object-fit: cover;
         }
diff --git a/react/features/filmstrip/constants.js b/react/features/filmstrip/constants.js
index 66c9ec65c..5355432ce 100644
--- a/react/features/filmstrip/constants.js
+++ b/react/features/filmstrip/constants.js
@@ -31,7 +31,7 @@ export const DISPLAY_DRAWER_THRESHOLD = 512;
 /**
  * Breakpoint past which a single column view is enforced in tile view.
  */
-export const SINGLE_COLUMN_BREAKPOINT = 300;
+export const SINGLE_COLUMN_BREAKPOINT = 800;
 
 /**
  * Breakpoint past which a two column view is enforced in tile view.
@@ -44,7 +44,7 @@ export const TWO_COLUMN_BREAKPOINT = 1000;
  * single column view.
  * If this is to be modified, please also change the related media query from the tile_view scss file.
  */
-export const ASPECT_RATIO_BREAKPOINT = 500;
+export const ASPECT_RATIO_BREAKPOINT = 100;
 
 /**
  * Minimum height of tile for small screens.
diff --git a/react/features/shared-video/components/web/AbstractVideoManager.js b/react/features/shared-video/components/web/AbstractVideoManager.js
index 43f7886b2..e16bf37f8 100644
--- a/react/features/shared-video/components/web/AbstractVideoManager.js
+++ b/react/features/shared-video/components/web/AbstractVideoManager.js
@@ -326,6 +326,7 @@ class AbstractVideoManager extends PureComponent<Props> {
      * @returns {void}
      */
     smartAudioMute() {
+        return;
         const { _isLocalAudioMuted, _muteLocal } = this.props;
 
         if (!_isLocalAudioMuted
