diff --git a/Makefile b/Makefile
index 2872b3485..9500ed346 100644
--- a/Makefile
+++ b/Makefile
@@ -128,5 +128,4 @@ source-package:
 	mkdir -p source_package/jitsi-meet/css && \
 	cp -r *.js *.html resources/*.txt favicon.ico fonts images libs static sounds LICENSE lang source_package/jitsi-meet && \
 	cp css/all.css source_package/jitsi-meet/css && \
-	(cd source_package ; tar cjf ../jitsi-meet.tar.bz2 jitsi-meet) && \
-	rm -rf source_package
+	(cd source_package ; tar cjf ../jitsi-meet.tar.bz2 jitsi-meet)
diff --git a/css/filmstrip/_tile_view.scss b/css/filmstrip/_tile_view.scss
index 083b4c083..7fee90ba6 100644
--- a/css/filmstrip/_tile_view.scss
+++ b/css/filmstrip/_tile_view.scss
@@ -97,7 +97,7 @@
      * Max-width corresponding to the ASPECT_RATIO_BREAKPOINT from features/filmstrip/constants,
      * from which we subtract the chat size.
      */
-    @media only screen and (max-width: calc(500px + #{$sidebarWidth})) {
+    @media only screen and (max-width: calc(100px + #{$sidebarWidth})) {
         video {
             object-fit: cover;
         }
diff --git a/react/features/filmstrip/constants.ts b/react/features/filmstrip/constants.ts
index 77484b998..a2e7e87a8 100644
--- a/react/features/filmstrip/constants.ts
+++ b/react/features/filmstrip/constants.ts
@@ -31,7 +31,7 @@ export const DISPLAY_DRAWER_THRESHOLD = 512;
  * single column view.
  * If this is to be modified, please also change the related media query from the tile_view scss file.
  */
-export const ASPECT_RATIO_BREAKPOINT = 500;
+export const ASPECT_RATIO_BREAKPOINT = 100;
 
 /**
  * Minimum height of tile for small screens.
diff --git a/react/features/shared-video/components/web/AbstractVideoManager.ts b/react/features/shared-video/components/web/AbstractVideoManager.ts
index 860155532..14823f25c 100644
--- a/react/features/shared-video/components/web/AbstractVideoManager.ts
+++ b/react/features/shared-video/components/web/AbstractVideoManager.ts
@@ -330,6 +330,7 @@ class AbstractVideoManager extends PureComponent<IProps> {
      * @returns {void}
      */
     smartAudioMute() {
+        return;
         const { _isLocalAudioMuted, _muteLocal } = this.props;
 
         if (!_isLocalAudioMuted
